rfcs/0093-rfc-categories.md

```diff
@@ -0,0 +1,152 @@
+---
+feature: rfc_categories
+start-date: 2021-05-19
+author: David Arnold
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: @kevnicox, @Mic92, @gytis-ivaskevicius
+shepherd-leader: @kevincox
+related-issues:
+category: process
+---
+
+# Summary
+[summary]: #summary
+
+The RFC process is amended with means to categorize RFCs into one of: _feature_, 
+_information_ or _process_, where each category sets different accents that
+improve the overall outcome of the process.
+
+# Motivation
+[motivation]: #motivation
+
+## Some issues are not addressed by the community
+
+There is no appropriate venue for it. When reading the current RFC process, it becomes
+clear that it was made primarily with purely technical decisions in mind. It might only
+be a perception, but this perception systemically skews what's being discussed and what
+not.
+
+As a result, important topics might not be adressed and settled by the community at large
+in a generally accepted procedure.
+
+Specifically, it is hard to propose a well coordinated experiment across the community,
+document and acknowledge design issues, record proof-generated insight, but also to
+amend the RFC process itself, the forum rules, the code of conduct or propose any other 
+binding changes to community workflows or infrastructure.
+
+**For example**, the [flake RFC](https://github.com/NixOS/rfcs/pull/49), which by many is
+considered a failed RFC process, might have benefited from a framework to transition into
+a general experiment in the form of an _informational_ RFC as soon as it had become
+clear that it won't be accepted as a _feature_ RFC. Its subsequent closure has rendered
+the entire flake experiment to a largely undocumented, unstructured, and intransparent
+process that still elicites strong opinions within the community. Clarity over RFC options
+and venues _might_ have helped mitigate this situation.
+
+## Some issues are not addressed from the appropriate angle
+
+Even if, in the past, people might have used the RFC Process to gather broader consensus
+around some of those hard-to-propose topics, the ensuing discussion still might have been
+framed in a way that is not best suited. This actually starts with the structure of the
+template.
+
+By explicitly categorizing RFCs, it will be immediatly evident for participants that
+those RFCs are a) legitimate and b) require an evaluation within the fair boundaries
+of their categories.
+
+**For example**, [RFC31](https://github.com/NixOS/rfcs/pull/31), a feature RFC, then was
+closed and superseded by [RFC46](https://github.com/NixOS/rfcs/pull/46), which ended up 
+being an informational RFC with certain inclination towards a process RFC, that was accepted.
+RFC46 is a prime example of the usefulness of "documenting language", as the RFC relates to
+its purpose. Having an explicit categorization is expected to suggest making use of these
+venues to a broder RFC author base.
```

Comment:
Since this was ultimately successful maybe it is worth explicitly claiming that the presence of categories upfront may have led this RFC in the right direction from the start.